### PR TITLE
Kim 163338392 office denies sit request

### PR DESCRIPTION
**Pull request by kimallen**
## Description

Allows office user to Deny a SIT request from a TSP.
This also does some refactoring of similar forms with unused state.

## Review Notes

Note that I have since changed the "Denied" to black after taking the giphy image
This also removes the 'Approve' and 'Deny' buttons in the Office app when a SIT is not in Requested status (see [thread](https://defensedigitalservice.slack.com/archives/G8HJRJBMZ/p1557441754186400))

## Setup
As office user, go to either SITAPR or SITDEN, go to the storage in transit panel, and click Deny icon and fill out form.

## Code Review Verification Steps

* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/163338392) for this change
* Also delivers [TSP story](https://www.pivotaltracker.com/story/show/164252427) for Denied SIT view
## Screenshots

SIT Deny

### CODE REVIEW

File: pkg/testdatagen/scenario/e2ebasic.go [hunk 1]
```diff
@@ -2286,6 +2286,53 @@ func (e e2eBasicScenario) Run(db *pop.Connection, loader *uploader.Uploader, log
...
+			TransportationServiceProviderID: tspUser.TransportationServiceProviderID,
+			Accepted:                        models.BoolPointer(true),
+		},
+	})
+
+	testdatagen.MakeStorageInTransit(db, testdatagen.Assertions{
+		StorageInTransit: models.StorageInTransit{
+			ShipmentID:         offer38.ShipmentID,
+			Shipment:           offer38.Shipment,
+			EstimatedStartDate: time.Date(2019, time.Month(3), 22, 0, 0, 0, 0, time.UTC),
```
Comment (jacquelineIO) [inline pkg/testdatagen/scenario/e2ebasic.go]: We are aiming to use `TestYear` in tests not sure if SIT is also effected by this as well. I don't know how SIT is calculated but to ensure we have valid dates for pricing, might want this to be `TestYear`. 
Thoughts?

File: src/shared/StorageInTransit/StorageInTransit.jsx [hunk 2]
```diff
@@ -114,6 +114,8 @@ export class StorageInTransit extends Component {
   render() {
     const { storageInTransit } = this.props;
     const { showTspEditForm, showOfficeEditForm, showApproveForm, showDenyForm, showPlaceInSitForm } = this.state;
+    const isDenied = storageInTransit.status === 'DENIED';
+    const isRequested = storageInTransit.status === 'REQUESTED';
```
Comment (jacquelineIO) [inline src/shared/StorageInTransit/StorageInTransit.jsx]: Setup constants here, but not using throughout the rest of the code?

File: src/shared/StorageInTransit/StorageInTransit.jsx [hunk 3]
```diff
@@ -123,15 +125,15 @@ export class StorageInTransit extends Component {
...
             </span>{' '}
-            {storageInTransit.status === 'REQUESTED' && <SitStatusIcon isTspSite={isTspSite} />}
+            {isRequested && <SitStatusIcon isTspSite={isTspSite} />}
           </span>
           {storageInTransit.status === 'APPROVED' ? (
             <span data-cy="storage-in-transit-status">
               <FontAwesomeIcon className="icon approval-ready" icon={faCheck} />
               Approved
             </span>
           ) : storageInTransit.status === 'DENIED' ? (
```
Comment (jacquelineIO) [inline src/shared/StorageInTransit/StorageInTransit.jsx]: update to const

File: src/shared/StorageInTransit/StorageInTransit.jsx [hunk 4]
```diff
@@ -157,15 +160,16 @@ export class StorageInTransit extends Component {
             )
           )}
           {showDenyForm ? (
-            <DenySitRequest onClose={this.closeDenyForm} />
+            <DenySitRequest onClose={this.closeDenyForm} storageInTransit={storageInTransit} />
           ) : storageInTransit.status === 'APPROVED' || storageInTransit.status === 'DENIED' ? (
```
Comment (jacquelineIO) [inline src/shared/StorageInTransit/StorageInTransit.jsx]: add new const and use it?

File: pkg/testdatagen/scenario/e2ebasic.go (hunk 1 above)
Comment (kimallen) [reply pkg/testdatagen/scenario/e2ebasic.go]: Looking into it. . .if I change those to TestYear, I'm not sure how to change the expected values to the test year in the e2e test.  I'm not finding any examples of dates expected based on TestYear in the e2e's.  Do you know how to do that?

File: src/shared/StorageInTransit/StorageInTransit.jsx (hunk 2 above)
Comment (kimallen) [reply src/shared/StorageInTransit/StorageInTransit.jsx]: Fixed

File: src/shared/StorageInTransit/StorageInTransit.jsx (hunk 3 above)
Comment (kimallen) [reply src/shared/StorageInTransit/StorageInTransit.jsx]: Done

File: src/shared/StorageInTransit/StorageInTransit.jsx (hunk 4 above)
Comment (kimallen) [reply src/shared/StorageInTransit/StorageInTransit.jsx]: Done

File: swagger/api.yaml [hunk 5]
```diff
@@ -3224,7 +3224,7 @@ paths:
           required: true
           description: UUID of storage in transit
         - in: body
-          name: storageInTransitApprovalPayload
+          name: storageInTransitDenyPayload
```
Comment (Ryan-Koch) [inline swagger/api.yaml]: Good catch on the naming mistake.

**Pull request merged**